include/benchmark/benchmark.h

```diff
@@ -1302,6 +1303,77 @@ struct SystemInfo {
   BENCHMARK_DISALLOW_COPY_AND_ASSIGN(SystemInfo);
 };
 
+// BenchmarkName wraps a std::string containing different fields
+// of a benchmark's name (e.g. the 'root' name, the benchmark's
+// 'args' etc.).
+class BenchmarkName {
+ public:
+  // Different fields of the benchmark's name.
+  enum Field {
```

Comment:
thinking more, and looking at the access pattern, i think this is overkill. The sections of the benchmark name are always appended in the same order. Similarly, you only care about dropping 'args' from the complexity bit (as far as i can see).

So why not just have a struct with all the strings stored as individual fields with an accessor `name` and another `NameWithoutArgs`.

Ie:

```c++
struct BenchmarkName {
  std::string name;
  std::string args;
  std::string min_time;
  std::string iterations;
  ...

  std::string name() { return absl::StrJoin({name, args, min_time, iterations...}, "/"); }  // or equivalent non-absl version
  std::string nameNoArgs() { return absl::StrJoin({name, min_time, iterations...}, "/"); }  // also non-absl.
};
```

then the benchmark name is stored (as you have) as a BenchmarkName and we have two simple accessors for getting the formatted version.

Reply:
That certainly makes the code simpler, I didn't consider it.

I guess the downsides of that are that in the common case (accessing the whole name) you have to re-build the full name every time. The current approach keeps the full name and can just return it. It also means adding a new field to the name requires a little bit more work than just adding an enumeration value (and, er, updating `enumerator_count`).

What do you think? Happy to go for the simpler option if preferred.